src/jaybenne/transport.cpp

```diff
@@ -111,20 +127,27 @@ TaskStatus TransportPhotons(MeshData<Real> *md, const Real t_start, const Real d
                                     "Particle initially outside X3 logical bnds!");
 
             // Calculate cell bounds
-            const Real xl = coords.Xc<parthenon::X1DIR>(ip) - 0.5 * dx_i;
-            const Real xu = coords.Xc<parthenon::X1DIR>(ip) + 0.5 * dx_i;
-            const Real yl = coords.Xc<parthenon::X2DIR>(jp) - 0.5 * dx_j;
-            const Real yu = coords.Xc<parthenon::X2DIR>(jp) + 0.5 * dx_j;
-            const Real zl = coords.Xc<parthenon::X3DIR>(kp) - 0.5 * dx_k;
-            const Real zu = coords.Xc<parthenon::X3DIR>(kp) + 0.5 * dx_k;
+            const Real xl = coords.template Xc<parthenon::X1DIR>(ip) - 0.5 * dx_i;
+            const Real xu = coords.template Xc<parthenon::X1DIR>(ip) + 0.5 * dx_i;
+            const Real yl = coords.template Xc<parthenon::X2DIR>(jp) - 0.5 * dx_j;
+            const Real yu = coords.template Xc<parthenon::X2DIR>(jp) + 0.5 * dx_j;
+            const Real zl = coords.template Xc<parthenon::X3DIR>(kp) - 0.5 * dx_k;
+            const Real zu = coords.template Xc<parthenon::X3DIR>(kp) + 0.5 * dx_k;
 
             // Extract physical quantities
             const Real &rho = vmesh(b, fjh::density(), kp, jp, ip);
             const Real &sie = vmesh(b, fjh::sie(), kp, jp, ip);
             const Real temp = eos.TemperatureFromDensityInternalEnergy(rho, sie);
             const Real &ff = vmesh(b, fj::fleck_factor(), kp, jp, ip);
-            const Real ss = scattering.TotalScatteringCoefficient(rho, temp, ee);
-            const Real aa = opacity.AbsorptionCoefficient(rho, temp, ee);
+            Real ss;
+            Real aa;
+            if constexpr (FT == FrequencyType::gray) {
+              ss = mscattering.RosselandMeanTotalScatteringCoefficient(rho, temp);
+              aa = mopacity.RosselandMeanAbsorptionCoefficient(rho, temp);
```

Comment:
Could this absorption use the option Rosseland option in `AbsorptionCoefficient`? Actually, should the mean scattering class have had an update to toggle between Rosseland and Planck means (sorry to have missed this if so)?

Reply:
Yeah youre right the general absorption option would be better here. Should the Rosseland and Planck named versions of this function just be deleted in singularity-opac?

Reply:
Will make the change here, at any rate, and maybe update `singularity-opac` to have the corresponding `ScatteringCoefficient(rho, temp, mode)` function.